### PR TITLE
Create admin APIs to list timer, visibility, transfer and replication tasks

### PR DESCRIPTION
**Pull request by feedmeapples**
**What changed?**

* Created an admin API that that allows to list Timer, Replication, Visibility and Transfer tasks

**Why?**

* simplifies listing timer, replication, visibility and transfer tasks in tctl admin command by removing the need for direct DB connection params

**How did you test it?**

```
./tctl admin shard list_tasks --shard_id 3 --task_type timer
              NAMESPACEID              |   WORKFLOWID    |                RUNID                 | TASKID  |      TASKTYPE       |            FIRETIME            | VERSION  
  3f8edc54-063c-4826-98ba-32844ef03352 | 27d82ad7###___5 | f66b1a35-23cd-4774-a8bc-81544744b0df | 1048652 | WorkflowTaskTimeout | 2021-11-22 21:08:21.275063605  |       0  
                                       |                 |                                      |         |                     | +0000 UTC                      |          
  3f8edc54-063c-4826-98ba-32844ef03352 | 047d7c9f###___2 | de43ce6a-e8d3-497e-b5d8-162b1910969b | 1048596 | ActivityTimeout     | 2021-11-22 21:26:31.658736606  |       0  
                                       |                 |                                      |         |                     | +0000 UTC                      |          
  3f8edc54-063c-4826-98ba-32844ef03352 | 047d7c9f###___3 | eaec9629-36a6-4e6d-933e-1433a27859df | 1048601 | ActivityTimeout     | 2021-11-22 21:26:31.667210198  |       0
...
```

./tctl admin shard list_tasks --task_type replication --shard_id 3

./tctl admin shard list_tasks --task_type transfer --shard_id 3

./tctl admin shard list_tasks --task_type visibility --shard_id 3

**Potential risks**

no major risks, mostly the new API itself 

**Is hotfix candidate?**

### CODE REVIEW

File: proto/internal/temporal/server/api/adminservice/v1/request_response.proto [hunk 1]
```diff
@@ -147,6 +147,68 @@ message GetShardResponse {
...
+    bytes next_page_token = 2;
+}
+
+message Task {
+    string namespace_id = 1;
+    string workflow_id = 2;
+    string run_id = 3;
+    int64 task_id = 4;
+    google.protobuf.Timestamp visibility_time = 5 [(gogoproto.stdtime) = true];
+    google.protobuf.Timestamp fire_time = 6 [(gogoproto.stdtime) = true];
```
Comment (wxing1292) [inline proto/internal/temporal/server/api/adminservice/v1/request_response.proto:208]: there should be one attributes, not 2

for timer tasks, fire time / visibility time is when the task should be processed
for non timer tasks, this time attribute should be time.Unix(0,0)

File: proto/internal/temporal/server/api/adminservice/v1/request_response.proto [hunk 2]
```diff
@@ -147,6 +147,68 @@ message GetShardResponse {
...
+    int64 version = 7;    
+}
```
Comment (wxing1292) [inline proto/internal/temporal/server/api/adminservice/v1/request_response.proto:210]: no task context? at least we should add task type information

File: proto/internal/temporal/server/api/adminservice/v1/request_response.proto (hunk 1 above)
Comment (feedmeapples) [reply proto/internal/temporal/server/api/adminservice/v1/request_response.proto:208]: 🙌

File: proto/internal/temporal/server/api/adminservice/v1/request_response.proto (hunk 2 above)
Comment (feedmeapples) [reply proto/internal/temporal/server/api/adminservice/v1/request_response.proto:211]: added task type

**Review by wxing1292 — APPROVED** [review]

**Pull request merged**